src/Gui/NavigationAnimation.cpp

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/****************************************************************************
+ *                                                                          *
+ *   Copyright (c) 2023 Bas Ruigrok (Rexbas) <rexbasie@gmail.com>           *
+ *                                                                          *
+ *   This file is part of FreeCAD.                                          *
+ *                                                                          *
+ *   FreeCAD is free software: you can redistribute it and/or modify it     *
+ *   under the terms of the GNU Lesser General Public License as            *
+ *   published by the Free Software Foundation, either version 2.1 of the   *
+ *   License, or (at your option) any later version.                        *
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#include "NavigationAnimation.h"
+#include <App/Application.h>
+#include "View3DInventorViewer.h"
+
+using namespace Gui;
+
+NavigationAnimation::NavigationAnimation()
+{
+    animationDuration = App::GetApplication()
+                            .GetParameterGroupByPath("User parameter:BaseApp/Preferences/View")
+                            ->GetFloat("AnimationDuration", 0.25f);
+}
+
+void NavigationAnimation::reset(const SoCamera* camera)
+{
+    targetOrientation = camera->orientation.getValue();
+    targetTranslation = SbVec3f(0, 0, 0);
+    totalTranslation = SbVec3f(0, 0, 0);
+}
+
+void NavigationAnimation::setDuration(float duration)
+{
+    animationDuration = duration;
+}
+
+void NavigationAnimation::setTargetOrientation(const SbRotation& orientation)
+{
+    targetOrientation = orientation;
+}
+
+void NavigationAnimation::translate(SbVec3f translation)
```

Comment:
The method name is inconsistent. For setting the orientation the method is setTargetOrientation, so this method should be called setTargetTranslation()